docs/source/wrappers.rst

```diff
@@ -0,0 +1,499 @@
+optimizer instance or optimizer class. If you pass a class,
+you will be able to specify it's arguments via parameter routing (see
+:ref:`param-routing`).
+
+batch_size
+^^^^^^^^^^
+
+This argument is passed to :py:func:`~tensorflow.keras.Model.fit`. See the the 
+`Keras Model docs`_ for more details.
+
+validation_split
+^^^^^^^^^^^^^^^^
+
+This argument is passed to :py:func:`~tensorflow.keras.Model.fit`. See the the 
+`Keras Model docs`_ for more details.
+
+callbacks
+^^^^^^^^^
+
+Single or list of callback instances or classes. Using classes will allow
+you to pass their parameters via parameter routing (see
+:ref:`param-routing`).
+For more information on Keras callbacks, see 
+`Keras Callbacks docs`_.
+
+metrics
+^^^^^^^
+
+List or dict of metrics. See the the 
+`Keras Metrics docs`_ for more details on using metrics.
+If you pass classes instead of string names or instances, 
+you will be able to pass their parameters via parameter routing (see
+:ref:`param-routing`).
+
+warm_start
+^^^^^^^^^^
+
+This argument determines whether each
+:py:func:`~scikeras.wrappers.BaseWrapper.fit` call leads to a re-initialization of
+the :py:class:`~scikeras.wrappers.BaseWrapper` or not. By default, when calling
+:py:func:`~scikeras.wrappers.BaseWrapper.fit`, the parameters of the net are
+initialized, so your previous training progress is lost (consistent
+with the sklearn ``fit()`` calls). In contrast, with
+``warm_start=True``, each :py:func:`~scikeras.wrappers.BaseWrapper.fit` call will
+continue from the most recent state.
+
+verbose
+^^^^^^^
+
+``False`` disables the progress bar and other logging
+while ``True`` enables it.
+This argument is passed to multiple methods of :py:class:`~tensorflow.keras.Model`.
+To set different values for ``fit`` and ``predict`` for example, you can use
+``fit__verbose=True`` and ``predict__verbose=False`` or
+``verbose=True`` and ``predict__verbose=False`` which would have the same effect
+since the non routed value from ``verbose=True`` would be passed to ``fit``.
+
+shuffle
+^^^^^^^
+
+This argument is passed to :py:func:`~tensorflow.keras.Model.fit`. See the the 
+`Keras Model docs`_ for more details.
+
+run_eagerly
+^^^^^^^^^^^
+
+This argument is passed to :py:func:`~tensorflow.keras.Model.fit`. See the the 
+`Keras Model docs`_ for more details.
+
+
+Methods of :py:class:`scikeras.wrappers.BaseWrapper`
+----------------------------------------------------
+
+fit(X, y, sample_weights=None)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is one of the main methods you will use. It contains everything
+required to train the model, be it batching of the data, triggering
+the callbacks, or handling the internal validation set.
+
+In general, we assume there to be an ``X`` and a ``y``. And if your
+task does not have an actual ``y``, you may pass ``y=None``.
+
+``X`` and ``y`` are expected to be array-like. SciKeras does not
+currently support :py:class:`tensorflow.data.Dataset` inputs.
```

Comment:
I might add a little more clarity to "array-like":

"The X and y arguments to `fit` are expected to by array-like and fit in memory (e.g, NumPy arrays). ...